File: src/main/java/nz/co/searchwellington/commentfeeds/detectors/GenericCommentFeedDetector.java
package nz.co.searchwellington.commentfeeds.detectors;

import java.net.URL;

public class GenericCommentFeedDetector implements CommentFeedDetector {
	
	private String regex;
	
	public GenericCommentFeedDetector() {		
	}

	public GenericCommentFeedDetector(String regex) {
		this.regex = regex;
	}

	public boolean isValid(URL url) {
		return url != null && url.toExternalForm().matches(regex);
	}
	
	public String getRegex() {
		return regex;
	}

	public void setRegex(String regex) {
		this.regex = regex;
	}

}

File: src/main/scala/nz/co/searchwellington/commentfeeds/detectors/GenericCommentFeedDetector.scala
package nz.co.searchwellington.commentfeeds.detectors

import java.net.URL
import java.util.regex.Pattern

class GenericCommentFeedDetector(regex: String) extends CommentFeedDetector {

  private val pattern = Pattern.compile(regex)

  override def isValid(url: URL) = url != null && pattern.matcher(url.toExternalForm).matches

}